trie/committer.go

```diff
@@ -55,34 +46,36 @@ var committerPool = sync.Pool{
 }
 
 // newCommitter creates a new committer or picks one from the pool.
-func newCommitter() *committer {
-	return committerPool.Get().(*committer)
+func newCommitter(owner common.Hash, collectLeaf bool) *committer {
```

Comment:
We should get rid of newCommitter, returnCommitterToPool and the pool itself. Originally the comitter had some internal hashers and buffers which made it relevant to reuse. That was dropped at some point but the pool never removed. There's nothing reused at this point across commits, so it;s just weird boilerplate.

Reply:
I mean, we can keep `newCommitter`, just have it create a new thing from scratch, not use a sync.Pool.